crates/rb-sys-test-helpers/src/ruby_test_executor.rs

```diff
@@ -22,25 +22,36 @@ pub struct RubyTestExecutor {
 
 impl RubyTestExecutor {
     pub fn start() -> Self {
-        let (sender, receiver) = mpsc::sync_channel::<Box<dyn FnOnce() + Send>>(1);
+        let (sender, receiver) = mpsc::sync_channel::<Box<dyn FnOnce() + Send>>(0);
 
-        // Spawn a new scoped thread
         let handle = thread::spawn(move || {
+            for closure in receiver {
+                closure();
+            }
+        });
+
+        let executor = Self {
+            sender: Some(sender),
+            handle: Some(handle),
+        };
+
+        executor.run(|| {
             static INIT: Once = Once::new();
+            static mut STATE: AtomicU8 = AtomicU8::new(0);
```

Comment:
Why is more synchronization needed now?  It looks like moving this initialization code into `executor.run(|| {` that precedes the executor being returned means that we shouldn't even need the `INIT`, let alone the extra `STATE` synchronization

Reply:
I was having some major synchronization headaches on Rust 1.54 (but not on stable). I threw everything at the book at it to finally make these tests stable since libruby doesn't always use atomic loads for these values. 

My thinking is that since these are test helpers, it's better have too much synchronization than not enough - to avoid flakiness at almost all costs. That being said, planning on circling back to clean up the unnecessary stuff.